### PR TITLE
Handle not finding any resource groups

### PR DESCRIPTION
**Pull request by ksnavely**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

I didn't run acceptance tests on this although local `make test` passes.

### CODE REVIEW

**Comment by ksnavely** [comment]
I ran into this when my account was unable to list the resource groups in another account used by our organization.

Terraform pukes:

```
panic: runtime error: index out of range [0] with length 0                                                                                                             
2021-04-13T20:13:45.589-0600 [DEBUG] plugin.terraform-provider-ibm_v1.21.1:                                                                                            
2021-04-13T20:13:45.589-0600 [DEBUG] plugin.terraform-provider-ibm_v1.21.1: goroutine 63 [running]:                                                                    
2021-04-13T20:13:45.589-0600 [DEBUG] plugin.terraform-provider-ibm_v1.21.1: github.com/IBM-Cloud/terraform-provider-ibm/ibm.dataSourceIBMResourceGroupRead(0xc000447a40
, 0x32ec380, 0xc000394e00, 0xc000447a40, 0x0)
...big fat stack trace
```

This should puke less I believe.

I'll review the contrib guidelines in detail.

File: ibm/data_source_ibm_resource_group.go [hunk 1]
```diff
@@ -64,6 +64,9 @@ func dataSourceIBMResourceGroupRead(d *schema.ResourceData, meta interface{}) er
 		if err != nil {
 			return fmt.Errorf("Error retrieving default resource group: %s", err)
 		}
+		if len(grp) < 1 {
```
Comment (hkantare) [inline ibm/data_source_ibm_resource_group.go:67]: Can you add a condition check if grp != nil && len(grp) <1

File: ibm/data_source_ibm_resource_group.go [hunk 2]
```diff
@@ -74,6 +77,9 @@ func dataSourceIBMResourceGroupRead(d *schema.ResourceData, meta interface{}) er
 		if err != nil {
 			return fmt.Errorf("Error retrieving resource group %s: %s", name, err)
 		}
+		if len(grp) < 1 {
```
Comment (hkantare) [inline ibm/data_source_ibm_resource_group.go:80]: similar like above

**Comment by hkantare** [comment]
closing this PR since we are handling with 
https://github.com/IBM-Cloud/terraform-provider-ibm/pull/2936

**Pull request closed without merging**